crates/backend/src/server/get_tokens.rs

```
@@ -170,7 +166,12 @@ impl RawMintRequest {
             .ok_or(MintRequestError::AssetAmount(self.asset_amount))?;
 
         // Check the API key, if provided
-        let api_key = self.api_key.as_deref().map(ApiKey::decode).transpose()?;
+        let api_key = self
+            .api_key
+            .as_deref()
+            .map(ApiKey::decode)
+            .transpose()
+            .map_err(|_| MintRequestError::InvalidApiKey(self.api_key.unwrap_or_default()))?;
```

If we end up ignoring the anyhow error to convert it back to `MintRequestError`, shouldn't we leave it as before? Where `ApiKey::decode` returned a `MintRequestError`

---

The API key is decoded in both the `pow` and `get_tokens` endpoints so I think it should return a generic error and be mapped on each endpoint. However, there was actually a naming error in the PoW endpoint so I fixed that in [734e149](https://github.com/0xMiden/miden-faucet/pull/10/commits/734e149d0b4903968ff2a1364039feeb20b22201)